technic/helpers.lua

```diff
@@ -63,23 +63,27 @@ function technic.swap_node(pos, name)
 end
 
 function technic.set_RE_charge(stack, charge)
-	local wear_factor = stack:get_definition().technic_wear_factor
-	if wear_factor then
-		local wear = math.floor(charge * wear_factor + 0.5)
+	local def = stack:get_definition()
+	if def.technic_set_charge then
+		def.technic_set_charge(stack, charge)
+	elseif def.technic_wear_factor then
+		local wear = math.floor(charge * def.technic_wear_factor + 0.5)
 		stack:set_wear(wear > 0 and 65536 - wear or 0)
 	else
 		minetest.log("error", "technic.set_RE_charge item not registered as power tool: "..stack:get_name())
 	end
 end
 
 function technic.get_RE_charge(stack)
-	local wear_factor = stack:get_definition().technic_wear_factor
-	if wear_factor then
+	local def = stack:get_definition()
+	if def.technic_get_charge then
+		return def.technic_get_charge(stack) or 0, 0
```

Comment:
Does this return the correct values? For example, if `def.technic_get_charge` returned `100, 100`, would it return `100, 0`?

Reply:
You're right. It does not return correct values and would be better written as
```lua
return (def.technic_get_charge(stack) or 0), 0
```
Should be fixed, upstream allows first value to be undefined and only guards second value which is max charge.
I guess it would be good to do that here as well, it also saves temporary tables for unpacking results.

Reply:
Actually there's no guards for user defined function at all if it is registered tool.